create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java

```diff
@@ -0,0 +1,121 @@
+package no.sikt.nva.scopus;
+
+import com.amazonaws.services.lambda.runtime.CognitoIdentity;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.net.HttpHeaders;
+import no.unit.nva.api.CreatePublicationRequest;
+import no.unit.nva.doi.fetch.model.Summary;
+import no.unit.nva.metadata.service.MetadataService;
+import no.unit.nva.model.EntityDescription;
+import nva.commons.apigateway.ApiGatewayHandler;
+import nva.commons.apigateway.GatewayResponse;
+import nva.commons.core.Environment;
+import nva.commons.core.JsonUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.util.Optional;
+
+import static org.apache.http.HttpHeaders.CONTENT_TYPE;
+import static org.apache.http.HttpStatus.SC_BAD_REQUEST;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasKey;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class ScopusHandlerTest {
```

Comment:
I think all the tests should start with testing handleRequest and nothing else. For anything else it is very difficult to follow the scenario. Example ```handler.processInput(null, null, null)``` would never occur.

Reply:
Maybe it might never occur, but JaCoco 100% demands it to be tested. 
You want the Scenarios be testet at Handler-level. But you need to test your functionality (that single classes provide) where it is provided. There you can test the behavior of the particular class.

Besides is `handler.processInput(null, null, null)` just a placeholder, until we have implemented something. But teh tight JaCoco and PMD rules forces us to write something up front.

Reply:
We will follow your test premisses.